bokeh/core/compat/__init__.py

```diff
@@ -24,3 +24,9 @@
 .. _mplexporter: https://github.com/mpld3/mplexporter/tree/master/mplexporter
 
 '''
+from ...util.deprecation import deprecated
+
+deprecated(
+    "MPL compatibility can no longer be successfully maintained, and is now deprecated. " +
+    "All MPL compat functions will be removed completely on the release of Bokeh 1.0. " +
```

Comment:
I'd be amendable to start removing the examples and docs before 1.0 but leaving the actual hard cutoff at for when then modules are removed at 1.0 There are not many places, I am confident that they can be removed fully in one go and if not, we can mop up any stragglers...